2-projects/1-todo-list/script.js

```diff
@@ -13,13 +45,43 @@ let todos = [
 populateTodoList(todos);
 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
+const addBtn = document.getElementsByClassName("btn btn-primary mb-3")[0];
+addBtn.addEventListener("click", addNewTodo);
+
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  // Write your code here... and remember to reset the input field to be blank aft er creating a todo!
+  let inputElement = document.getElementById("todoInput");
+  if (inputElement.value === "") {
+    
+  } else {
+    let ulFlusher = document.getElementById("todo-list");
+    ulFlusher.innerHTML = "";
+    let newItem = {};
+    newItem["task"] = inputElement.value;
+    newItem["completed"] = false;
+    todos.push(newItem);
+    populateTodoList(todos);
+  }
 }
 
+
+
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
+  let listEle = document.getElementById("todo-list");
+  const liEle = document.querySelectorAll("li");
+  for (let li of liEle) {
+    if (li.style.textDecoration === "line-through") {
+      listEle.removeChild(li);
+    }
+  }
 }
+const delBtnComp = document.getElementById("remove-all-completed");
+delBtnComp.addEventListener("click", deleteAllCompletedTodos);
+
+    
+
+
```

Comment:
Great job! Just remember the annoying default behaviour of a button when placed inside a form - it will refresh the page. I think you already figured out how to stop this in a previous exercise... 😏